server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -172,10 +180,14 @@ public int getNumOpenConnections()
         private final AtomicLong byteCount = new AtomicLong(0);
         private final BlockingQueue<InputStream> queue = new LinkedBlockingQueue<>();
         private final AtomicBoolean done = new AtomicBoolean(false);
+        private final AtomicReference<String> fail = new AtomicReference<>();
 
         @Override
         public ClientResponse<InputStream> handleResponse(HttpResponse response)
         {
+          checkQueryTimeout();
```

Comment:
I think it would be nicer if both of these were to return booleans and the call sites were to just skip things or whatever if the limits are exceeded.